controllers/fenceagentsremediation_controller_test.go

```diff
@@ -208,16 +208,68 @@ var _ = Describe("FAR Controller", func() {
 			})
 		})
 	})
+	Context("Reconcile with OutOfServiceTaint strategy", func() {
+		nodeKey := client.ObjectKey{Name: workerNode}
+		farNamespacedName := client.ObjectKey{Name: workerNode, Namespace: defaultNamespace}
+		farNoExecuteTaint := utils.CreateFARNoExecuteTaint()
+		outOfServiceTaint := utils.CreateOutOfServiceTaint()
+		conditionStatusPointer := func(status metav1.ConditionStatus) *metav1.ConditionStatus { return &status }
+		BeforeEach(func() {
+			farPod := createRunningPod("far-manager-test", farPodName, "")
+			DeferCleanup(deletePod, farPod)
+		})
+		JustBeforeEach(func() {
+			// Create node, and at the end clean it up with DeferCleanup
+			Expect(k8sClient.Create(context.Background(), node)).To(Succeed())
+			DeferCleanup(k8sClient.Delete, context.Background(), node)
+		})
+
+		When("creating valid FAR CR", func() {
+			BeforeEach(func() {
+				node = utils.GetNode("", workerNode)
+				underTestFAR = getFenceAgentsRemediation(workerNode, fenceAgentIPMI, testShareParam, testNodeParam, v1alpha1.OutOfServiceTaintRemediationStrategy)
+				Expect(k8sClient.Create(context.Background(), underTestFAR)).To(Succeed())
+			})
+			It("should have both far taint and out-of-service taint, and at the end they will be deleted", func() {
+				By("Searching for remediation taint")
+				Eventually(func(g Gomega) bool {
+					g.Expect(k8sClient.Get(context.Background(), nodeKey, node)).To(Succeed())
+					g.Expect(k8sClient.Get(context.Background(), farNamespacedName, underTestFAR)).To(Succeed())
+					res, _ := cliCommandsEquality(underTestFAR)
+					return utils.TaintExists(node.Spec.Taints, &farNoExecuteTaint) && utils.TaintExists(node.Spec.Taints, &outOfServiceTaint) && res
+				}, timeoutFinalizer, pollInterval).Should(BeTrue(), "both far taint and out-of-service taint should be added, and command format is correct")
+
```

Comment:
I think it doesn't make sense for the OutOfServiceTaint remediation.
To get the effect of out-of-service taint, the Pod GC controller, part of kube-controller-manager, should be executed.
But the envtest framework doesn't run them and the test pod won't be deleted by the out-of-service taint. 
So, we can't check if the taint works well.

As for the ResourceDeletion remediation, it tries to delete the test pod manually as part of the logic of FAR. So it will make sense to test if pods were deleted with verifyPodDeleted().

Reply:
> But the envtest framework doesn't run them

ah yes, of course, I probably mixed unit and e2e test, sorry 👍🏼